### DISCUSSION TITLE
Navidrome and updating metadata: What's your workflow?

### DISCUSSION THREAD

**Discussion by celynw** [beetbox/beets#5372] [General]
Calling all Navidrome users: I'd like to know what your workflow is!

Navidrome relies on the path of a track being consistent for tracking data like the last played date and the user rating. The reason for this is that by design, Navidrome will never modify the audio files themselves.

The issue comes when beets moves that file, Navidrome. This is hard to avoid! Depending on your path format, any change to the title, album name, track number, will cause Navidrome to treat it as a new track.

How do you deal with this?

I am seriously considering changing my path format to this:

```yml
paths:
  default: $album_id/$id
  singleton: Singleton/$id
```

I need the album subdirectory so I can store the cover art as separate files (`cover.png`).

Although, if I wish to upgrade my lossy files to lossless, the file extension will change. I could write a small plugin/tool to update the extensions (definitely much simpler than writing a plugin to handle entire path changes), but I'm wondering if there's a better way. Omit the extensions altogether??

**Comment by michaelkrieger** [comment]
You may want to wait out for https://github.com/navidrome/navidrome/pull/2709 which aims to fix this issue https://github.com/navidrome/navidrome/issues/2068 . This likely won't handle if you replace/upgrade a release, but should if you move things around.

You could also make a hook ( https://beets.readthedocs.io/en/v1.4.7/plugins/hook.html ) on the item_moved event that simply updates Navidrome's database via the API or via queuing up a SQLite command to point to the new file.
